### ISSUE TITLE
[Bug]: Configuration row arguments may get misplaced between Python SchemaTransformPayload encoding and  Java RowCoder decoding

### ISSUE THREAD

**Issue by ahmedabu98** [apache/beam#25669]
### What happened?

Was testing a SchemaTransform Python wrapper (#25521) and found that I had to have a right ordering of kwargs for the input arguments to reach the Java transform in the right fields. This is weird because the ordering of kwargs should have no impact.

For example, where `self._table="my_project:my_dataset.xlang_table"`,

the following works fine:
```
external_storage_write = SchemaAwareExternalTransform(
    identifier=self.schematransform_config.identifier,
    expansion_service=self._expansion_service,
    createDisposition=self._create_disposition,
    writeDisposition=self._write_disposition,     #<---
    triggeringFrequencySeconds=self._triggering_frequency,
    useAtLeastOnceSemantics=self._use_at_least_once,
    table=self._table)                            #<---
```
and I get a configuration object in Java transform that looks like this:
```
BigQueryStorageWriteApiSchemaTransformConfiguration{
  table=my_project:my_dataset.xlang_table, 
  createDisposition=, 
  writeDisposition=, 
  triggeringFrequencySeconds=0, 
  useAtLeastOnceSemantics=false}
```

However, if I change the kwargs to look like this (switch places of table and writeDisposition):
```
external_storage_write = SchemaAwareExternalTransform(
    identifier=self.schematransform_config.identifier,
    expansion_service=self._expansion_service,
    createDisposition=self._create_disposition,
    table=self._table,                            #<---
    triggeringFrequencySeconds=self._triggering_frequency,
    useAtLeastOnceSemantics=self._use_at_least_once,
    writeDisposition=self._write_disposition)     #<---
```
I get the following configuration object. Notice the value intended for `table` is now in the `writeDisposition` field.
```
BigQueryStorageWriteApiSchemaTransformConfiguration{
  table=, 
  createDisposition=, 
  writeDisposition=my_project:my_dataset.xlang_table, 
  triggeringFrequencySeconds=0, 
  useAtLeastOnceSemantics=false
```

### Issue Priority

Priority: 1 (data loss / total loss of function)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by ahmedabu98** [comment]
I followed this [unittest](https://github.com/apache/beam/blob/ddae966f3346fbe247486324cbf8a8a532895316/sdks/python/apache_beam/transforms/external_test.py#L449) and tried encoding my configuration with SchemaTransformPayloadBuilder() and decoding with proto_utils.parse_Bytes(). It returned a configuration with fields in the right order.

Not sure what the mismatch is between how Python and Java handle the payload.

**Comment by Abacn** [comment]
Observed similar issues in #26480. The fix of #25685 appears to be generic to schema but the same still happens for SchemaIO wrapper (used by jdbc xlang).

**Comment by chamikaramj** [comment]
jdbc.py has a similar issue but that's not addressed here. transforms in jdbc.py needs to preserve ordering since it uses a second config object [1] along with (old) SchemaIO API. This config object that is encoded in the Python side and decoded in the Java side. So ordering of the the objects in the two sides have to match.

One way to fix this will be to update jdbc.py to use schema-aware transforms (which will pick up the fix provided in this PR).

[1] https://github.com/apache/beam/blob/e868c8da79a8cf727044cf805b2041fbdb8ff181/sdks/python/apache_beam/io/jdbc.py#L339
